port/port_config.h.in

```diff
@@ -15,6 +15,11 @@
 #cmakedefine01 HAVE_FULLFSYNC
 #endif  // !defined(HAVE_FULLFSYNC)
 
+// Define to 1 if you have a definition for O_CLOEXEC in <fcntl.h>.
+#if !defined(HAVE_O_CLOEXEC)
+#cmakedefine01 HAVE_O_CLOEXEC
```

Comment:
[`#cmakedefine01`](https://cmake.org/cmake/help/v3.9/command/configure_file.html) always defines the `HAVE_O_CLOEXEC` macro:
> Input file lines of the form `#cmakedefine01 VAR` will be replaced with either `#define VAR 1` or `#define VAR 0` similarly.

Therefore, it looks wrong to use `#if defined(HAVE_O_CLOEXEC)` in the code: https://github.com/google/leveldb/blob/75fceae7003e217e16b04433831da7528ae56881/util/env_posix.cc#L52 and https://github.com/google/leveldb/blob/75fceae7003e217e16b04433831da7528ae56881/util/env_posix_test.cc#L62